From 24214ca5d506f56dc0cb2a2e1312256472039475 Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Mon, 15 Nov 2010 08:07:53 +0000
Subject: tar: fix --verify option, which broke in 1.24

* src/compare.c (verify_volume): Decode the header before invoking
diff_archive, as diff_archive no longer does this as of the
2010-06-28 commit.  Also, don't try to invoke diff_archive on a
zero block.
---
diff --git a/src/compare.c b/src/compare.c
index 6b7e6d8..f3112c6 100644
--- a/src/compare.c
+++ b/src/compare.c
@@ -611,8 +611,10 @@ verify_volume (void)
 		       (0, 0, _("A lone zero block at %s"),
 			STRINGIFY_BIGINT (current_block_ordinal (), buf)));
             }
+	  continue;
 	}
 
+      decode_header (current_header, &current_stat_info, &current_format, 1);
       diff_archive ();
       tar_stat_destroy (&current_stat_info);
     }
